src/Grassmann.jl

```diff
@@ -0,0 +1,263 @@
+using LinearAlgebra: svd, qr, Diagonal, det
+import LinearAlgebra: norm
+@doc doc"""
+    Grassmann{n,k} <: Manifold
+
+The Grassmann manifold $\operatorname{Gr}(n,k)$ consists of all $k$-dimensional
+subspaces of $\mathbb F^n$, where $\mathbb F \in \{\mathbb R, \mathbb C\}$ is
+either the real- (or complex-) valued subspaced of the $n$-dimensional (complex)
+space.
+
+The manifold can be represented as
+````math
+\operatorname{Gr}(n,k) \coloneqq \bigl\{ \operatorname{span}(x)
+: x \in \mathbb F^{n\times k}, \bar{x}^\mathrm{T}x = I_k\},
+````
+where $\bar\cdot$ denotes the complex conjugate and $I_k$ is the $k\times k$
```

Comment:
I haven't worked with Grassman manifolds before, is this actually an isometric embedding? It doesn't look like it but `inner` is just `dot`?

Reply:
Grassmann is a quotient of Stiefel, so you can use the inner product and it is a sub manifold of R^n (C^n) and inherits the inner product. Due to the quotient I am not sure about the isometry.

Reply:
AFAIK that's only true when the quotient has a global isometric section and we use it. In general we need horizontal-vertical decomposition of tangent vectors (see Section 1.5 of https://pdfs.semanticscholar.org/aad1/de01cdea203a3bb25db0cd0e45a611db23fe.pdf for example).

Reply:
Unless you are careful enough to only store the horizontal part at all times?

Reply:
No I am not that careful but the formulae are invariant to that part, so while it might not be equal element wise. For example any element from the horizontal part of the zero tangent vector still yields a norm of zero.

Reply:
I think I see what happens. `log` is constructed carefully enough to make `dot` the correct inner product. Tangent vectors are tied to a particular representation of the point but no one actually expects anything more. I did my quotient manifolds a bit differently but this might be a better approach, at least when it can be used.